openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -1449,85 +1455,111 @@ impl<T: CpuIo> X86EmulatorSupport for UhEmulationState<'_, '_, T, SnpBacked> {
         self.vp.partition.caps.vendor
     }
 
-    fn state(&mut self) -> Result<x86emu::CpuState, Self::Error> {
+    fn gp_sign_extend(&mut self, reg: usize) -> i64 {
+        self.gp(reg) as i64
+    }
+
+    fn gp(&mut self, reg: usize) -> u64 {
```

Comment:
We should make these values an enum and just pass that in so that we can remove the _ branch in the match.

Reply:
I don't like the panic catch-all either, but I'm not seeing how an enum solves the problem.

e.g. we could make `RegisterIndex` an enum, but then we'd just be moving the catch-all into `impl Into<RegisterIndex> for Register`.

I have similar thinking for the other enum conversions you're suggesting, am I missing something?